java-library-template/build.gradle.kts

```
@@ -14,14 +24,10 @@ object Meta {
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_21
-}
-
-plugins {
-    // Apply the java-library plugin for API and implementation separation.
-    `java-library`
-    `maven-publish`
-    signing
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(21))
+        vendor.set(JvmVendorSpec.AMAZON)
```

```suggestion
        languageVersion = JavaLanguageVersion.of(21)
        vendor = JvmVendorSpec.AMAZON
```
Cosmetic only, but I think it would align better with the DSL used in this file in other places.

---

Fixed ✔️ 
https://github.com/thriving-dev/java-library-template/pull/32/commits/43825452b9f2b9be60fffa1863fb8e68b25d4f8f